docs/usage.md

```diff
@@ -0,0 +1,212 @@
+from the Linux kernel. Its main advantage is that it does not build the
+entire kernel, only the files containing functions from `SYMBOL_LIST`. The
+kernel source to build must be properly configured (by `make prepare`) and
+all the tools necessary for building kernel must be installed. It also
+requires the `cscope` tool to be installed.
+
+#### Comparing sysctl options
+
+DiffKemp supports also a comparison of semantics of sysctl options. The list
+of the options to compare can be passed via `SYMBOL_LIST` to the `build-kernel`
+command. In such case, use `--sysctl` switch to generate snapshot for sysctl
+parameter comparison. The `compare` command is used in normal way.
+
+Sysctl option comparison compares the semantics of the proc handler function and
+semantics of all functions using the data variable that the sysctl option sets.
+
+It is possible to use patterns to specify a number of multiple sysctl options at
+once such as:
+
+- `kernel.*`
+- `kernel.{sysctl-1|sysctl-2}`
+
+Currently, these sysctl option groups are supported: `kernel.*`,
+`vm.*`, `fs.*`, `net.core.*`, `net.ipv4.conf.*`.
+
+#### Options
+
+- `KERNEL_DIR`: Path to a kernel's root directory.
+- `SNAPSHOT_DIR`: Output directory for storing the created snapshot.
+- `SYMBOL_LIST`: Path to a file containing a list of symbols (each symbol on
+  a single line) to be prepared for comparison.
+  In case `--sysctl` is used, the list is interpreted as a list of sysctl
+  parameters.
+- `--sysctl`: Compares sysctl option.
+- `--no-souce-dir`: Does not store the path to the source kernel directory in
+  the snapshot.
+
+### c) `llvm-to-snapshot`: snapshot generation from a single LLVM IR file
+
+```sh
+diffkemp llvm-to-snapshot PROJ_DIR LLVM_FILE SNAPSHOT_DIR SYMBOL_LIST
+```
+
+This command can be used if the project is already compiled into a single
+LLVM IR file.
+The file name is given in `LLVM_FILE` and must be relative to `PROJ_DIR`.
+
+#### Options
+
+- `PROJ_DIR`: Path to a project's root directory.
+- `LLVM_FILE`: Path to the LLVM IR file relative to the project's root
+  directory.
+- `SNAPSHOT_DIR`: Output directory for storing the created snapshot.
+- `SYMBOL_LIST`: Path to a file containing a list of symbols (each symbol on
+  a single line) which should be prepared for comparison.
+
+## 2. Semantic comparison command
+
+```sh
+diffkemp compare SNAPSHOT_DIR_1 SNAPSHOT_DIR_2
+```
+
+This command takes two snapshots and compares them for semantic equality.
+
+Syntactic diffs of the discovered differences are stored in separate files
+(one file for each compared function that is different) in a newly created
+directory. The name of the directory can be specified using the `-o` option,
+otherwise it is generated automatically.
+
+### Options
+
+- `SNAPSHOT_DIR_1`, `SNAPSHOT_DIR_2`: Paths to directories containing snapshots
+  of the project's version to be compared.
+- `--output-dir OUTPUT_DIR`, `-o OUTPUT_DIR`: Name of the output directory.
```

Comment:
I think this is understandable and shorter:
```suggestion
- `-o`, `--output-dir OUTPUT_DIR`: Name of the output directory.
```
Same for the other options below which have both the short and the long variant.